lib/pbench/test/unit/server/conftest.py

```diff
@@ -96,31 +93,43 @@ def setup(request, pytestconfig):
 
     cfg_file = pbench_cfg / "pbench-server.cfg"
     with cfg_file.open(mode="w") as fp:
-        fp.write(server_cfg_tmpl.format(TMP=TMP.name))
+        fp.write(server_cfg_tmpl.format(TMP=str(tmp_d)))
+
+    return dict(tmp=tmp_d, cfg_file=str(cfg_file))
 
-    pytestconfig.cache.set("TMP", TMP.name)
-    pytestconfig.cache.set("_PBENCH_SERVER_CONFIG", str(cfg_file))
 
-    def teardown():
-        """Test package teardown for pbench-server"""
-        TMP.cleanup()
+@pytest.fixture(scope="session", autouse=True)
```

Comment:
Is there a point to the `autouse=True` if it's got a *value* that's required by dependent fixtures? To get the value you have to reference it in the fixture signature anyway. Or are there fixtures I missed that simply expect to find the file on disk? (And, if so, why don't they all work that way instead?)

Or to put it another way, if a JSON file falls in the test case and nobody's there to read it, does it make a sound?

Reply:
Since `on_disk_server_config` is a "session" scoped fixture, it is run once before a test thread starts executing tests.  But, other fixtures, see `server_config` below, rely on its data structure to find the `cfg_file` that is setup for tests.

Reply:
Sure; but the root question is that if the *value* is important to tests that depend on this fixture, then it has to be referenced specifically by that test or some fixture used by the test; so what is the `autouse` buying us? Is the setup here important for tests that don't reference this fixture or any fixture that references it? If so, then a comment might be in order. If not, then it would seem that the `autouse` is serving no purpose. (Though I approved on the basis that it probably doesn't hurt, it may somewhat obfuscate the actual scope and dependencies.)

Reply:
Not sure `autouse` matters here since `server_config` is pretty much used all over the place, and removing the `autouse` parameter did nothing to affect the unit tests.

I think the `server_config` could be marked `autouse`, and its references removed except for the tests that need the value.

But that seems like a separate piece of work.